java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/MavenBuildStep.java

```diff
@@ -53,7 +53,7 @@ protected void doBuild(Path directory, Map<String, String> metadata) throws Buil
 
   @VisibleForTesting
   String getMavenHome() {
-    return System.getenv("M2_HOME");
+    return System.getenv("MAVEN_HOME");
```

Comment:
The new base image sets `MAVEN_HOME` and not `M2_HOME`, so this needed to change as well. I agree, probably not the most important thing.